### PR TITLE
[ADD][8.0]Website product supplier

### PR DESCRIPTION
**Pull request by sergio-teruel**
Module that allow a supplier (Manufaturer) create products

### CODE REVIEW

File: website_product_supplier/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+{
+    'name': "Website Product Supplier",
+    'category': 'Website',
+    'version': '8.0.1.0.0',
+    'depends': [
+        'auth_supplier',
+        'website_sale',
+        'website_portal',
+        'product_manufacturer'
```
Comment (rafaelbn) [inline website_product_supplier/__openerp__.py]: Why depends on product_manufacturer? It mustn't as you have suppliers without manufacturers. Better not use this dependency.

File: website_product_supplier/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+# (c) 2015 Antiun Ingeniería S.L. - Sergio Teruel
+# (c) 2015 Antiun Ingeniería S.L. - Carlos Dauden
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+{
+    'name': "Website Product Supplier",
+    'category': 'Website',
+    'version': '8.0.1.0.0',
+    'depends': [
+        'auth_supplier',
```
Comment (rafaelbn) [inline website_product_supplier/__openerp__.py]: Where is this module?

File: website_product_supplier/__openerp__.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        'website_sale',
```
Comment (rafaelbn) [inline website_product_supplier/__openerp__.py:11]: Why is needed website_sale? I think that suppliers could add products to be sold offline.

**Comment by rafaelbn** [comment]
@sergio-incaser please review test, runbot is failing

**Comment by sergio-teruel** [comment]
@rafaelbn this module depends of auth_supplier that has in another PR and another repo... so... :open_mouth:

**Comment by rafaelbn** [comment]
Please refer in this dependency in this PR to check it, thanks

File: website_product_supplier/__openerp__.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+{
+    'name': "Website Product Supplier",
+    'category': 'Website',
+    'version': '8.0.1.0.0',
+    'depends': [
+        'auth_supplier',
+        'website_portal',
+        'product_manufacturer'
```
Comment (rafaelbn) [inline website_product_supplier/__openerp__.py]: Please do not use product_manufacturer, user Odoo core functionality: product.supplierinfo

Comment (sergio-teruel) [inline website_product_supplier/__openerp__.py]: @rafaelbn And if i add more than one supplier, i have more than one owner for the same product and this is not correct.

Comment (rafaelbn) [inline website_product_supplier/__openerp__.py]: Not if this module restrict that you can have only one.

File: website_product_supplier/__openerp__.py (hunk 2 above)
Comment (yajo) [inline website_product_supplier/__openerp__.py]: Pending to be merged: https://github.com/OCA/server-tools/pull/277

**Comment by yajo** [comment]
https://github.com/OCA/server-tools/pull/277 is merged. Please rebase or merge.

**Comment by sergio-teruel** [comment]
Rebase done after OCA/server-tools#277 merged but dependency error continue

**Comment by rafaelbn** [comment]
cc @pedrobaeza could you help us with dependency?

**Comment by pedrobaeza** [comment]
Same as the other PR

**Comment by yajo** [comment]
Local tests show error 500 on wrong input, duplicated entry under right-side menu "My account" and "My supplier account" (both lead to the same place).

Also, no way to publish the product, I don't know if it is intended...

**Comment by rafaelbn** [comment]
I cannot test it, after registering I get:

```
Internal Server Error

The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.
```

Please check

**Comment by rafaelbn** [comment]
ping @sergio-incaser

File: website_product_supplier/README.rst [hunk 5]
```diff
@@ -0,0 +1,61 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+    :alt: License: AGPL-3
+
+========================
+Website Product Supplier
+========================
+
+This module was written to allow suppliers to create new products in frontend.
```
Comment (rafaelbn) [inline website_product_supplier/README.rst]: s/in frontend/in website_portal_purchase (we must fix this)

File: website_product_supplier/__openerp__.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+# (c) 2015 Antiun Ingeniería S.L. - Carlos Dauden
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+{
+    'name': "Website Product Supplier",
+    'category': 'Website',
+    'version': '8.0.1.0.0',
+    'depends': [
+        'auth_supplier',
+        'website_portal',
```
Comment (rafaelbn) [inline website_product_supplier/__openerp__.py]: It must depends in website_portal_purchase only, this is a new functionality only for suppliers

File: website_product_supplier/__openerp__.py [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    'data': [
+        'security/ir.model.access.csv',
+        'views/website_product_supplier.xml',
+        'views/product_supplier_view.xml',
+        'security/website_product_supplier_security.xml',
+    ],
+    'author': 'Antiun Ingeniería S.L., '
+              'Incaser Informatica S.L., '
+              'Odoo Community Association (OCA)',
+    'website': 'http://www.incaser.es',
```
Comment (rafaelbn) [inline website_product_supplier/__openerp__.py]: www.antiun.com

File: website_product_supplier/__openerp__.py [hunk 8]
```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+# (c) 2015 Antiun Ingeniería S.L. - Sergio Teruel
+# (c) 2015 Antiun Ingeniería S.L. - Carlos Dauden
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+{
+    'name': "Website Product Supplier",
+    'category': 'Website',
+    'version': '8.0.1.0.0',
+    'depends': [
```
Comment (rafaelbn) [inline website_product_supplier/__openerp__.py:10]: If you are writing in model  product.supplierinfo you must add dependency to module product. Isn't it?

Comment (sergio-teruel) [inline website_product_supplier/__openerp__.py:10]: **Dependency list:**

_website_portal_purchase_ :arrow_right: _purchase_ :arrow_right: _stock_account_ :arrow_right: _stock_ :arrow_right: _product_
So it is not necessary.

File: website_product_supplier/__openerp__.py [hunk 9]
```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+# (c) 2015 Antiun Ingeniería S.L. - Sergio Teruel
+# (c) 2015 Antiun Ingeniería S.L. - Carlos Dauden
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+{
+    'name': "Website Product Supplier",
```
Comment (yajo) [inline website_product_supplier/__openerp__.py:7]: Shouldn't this module be called `website_portal_product_supplier`? @pedrobaeza

File: website_product_supplier/__openerp__.py [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+    'category': 'Website',
+    'version': '8.0.1.0.0',
+    'depends': [
+        'website_portal_purchase',
```
Comment (yajo) [inline website_product_supplier/__openerp__.py:12]: Where's this module?

Comment (yajo) [inline website_product_supplier/__openerp__.py:12]: Autoanswer: https://github.com/OCA/website/pull/90/

File: website_product_supplier/views/product_supplier_view.xml [hunk 11]
```diff
@@ -0,0 +1,204 @@
...
+                                        <section id="product_fields_base">
+                                            <div id="product_name" class="row">
+                                                <div t-attf-class="col-md-8 col-xs-12 form-group #{error.get('product_name') and 'has-error' or ''}">
+                                                    <label class="control-label">Name</label>
+                                                        <input id="product_name" class="form-control" name="product_name" placeholder="Product name..."
+                                                               t-att-value="supplierinfo.get('product_name')"/>
+                                                </div>
+                                                <div t-attf-class="col-md-4 col-xs-12 form-group #{error.get('product_code') and 'has-error' or ''}">
+                                                    <label class="control-label">Code</label>
+                                                        <input id="product_code" class="form-control" name="product_code" placeholder="Product Code..."
```
Comment (yajo) [inline website_product_supplier/views/product_supplier_view.xml]: Placeholders should indicate examples, not repeat the field label.

For example, this field is labeled "Code", and the placeholder could be something like "123456".

The same applies for all other fields.

File: website_product_supplier/views/product_supplier_view.xml [hunk 12]
```diff
@@ -0,0 +1,204 @@
...
+                                                               t-att-value="supplierinfo.get('product_code')"/>
+                                                </div>
+                                            </div>
+                                            <div id="supplier_info" class="row">
+                                                <div t-attf-class="col-md-4 col-xs-12 form-group #{error.get('min_qty') and 'has-error' or ''}">
+                                                    <label class="control-label">Min Quantity</label>
```
Comment (yajo) [inline website_product_supplier/views/product_supplier_view.xml:104]: "Minimal Quantity"

File: website_product_supplier/views/product_supplier_view.xml [hunk 13]
```diff
@@ -0,0 +1,204 @@
...
+                                                    <div class="controls">
+                                                        <input id="min_qty" class="form-control" name="min_qty" placeholder="Min Quantity..."
+                                                               t-att-value="supplierinfo.get('min_qty')"/>
+                                                    </div>
+                                                </div>
+                                                <div t-attf-class="col-md-4 col-xs-12 form-group #{error.get('delay') and 'has-error' or ''}">
+                                                    <label class="control-label">Delay</label>
```
Comment (yajo) [inline website_product_supplier/views/product_supplier_view.xml:111]: "Delay" is not very easy to understand by itself. Please use something more descriptive, or use some tooltips for those terms not easy to understand.

File: website_product_supplier/views/product_supplier_view.xml [hunk 14]
```diff
@@ -0,0 +1,204 @@
...
+                                                    <div class="controls">
+                                                        <input id="delay" class="form-control" name="delay" placeholder="Delay..."
+                                                               t-att-value="supplierinfo.get('delay')"/>
+                                                    </div>
+                                                </div>
+                                            </div>
+                                            <div id="price_list" class="row">
+                                                <div class="col-md-4 col-xs-12 form-group">
+                                                    <label class="control-label">Price Min Quantity</label>
```
Comment (yajo) [inline website_product_supplier/views/product_supplier_view.xml:120]: "Price for Minimal Quantity"

File: website_product_supplier/views/product_supplier_view.xml [hunk 15]
```diff
@@ -0,0 +1,204 @@
...
+                                                    <div class="controls">
+                                                        <input id="min_quantity" class="form-control" name="min_quantity" placeholder="Min Quantity..."
+                                                               t-att-value="pricelist and pricelist[0].min_quantity"/>
+                                                    </div>
+                                                </div>
+                                                <div class="col-md-4 col-xs-12 form-group">
+                                                    <label class="control-label">Price</label>
```
Comment (yajo) [inline website_product_supplier/views/product_supplier_view.xml:127]: It's not clear the difference between Price and Price Min Qty. See comments above.

**Comment by rafaelbn** [comment]
Please @sergio-incaser try to review this PR, we must try to review againg, fix runbot and travis, thanks

**Comment by rafaelbn** [comment]
This is a WIP until https://github.com/OCA/website/pull/90 is merged

**Comment by rafaelbn** [comment]
Please @sergio-incaser @carlos-incaser rebase and we will see what's going on with this PR once #90 is merged, thanks

**Comment by sergio-teruel** [comment]
@rafaelbn Rebase done

**Comment by yajo** [comment]
After rebase, you are changing 78 files. It seems something went wrong when rebasing, please review it.

**Comment by rafaelbn** [comment]
Hi @sergio-incaser ,

Tested. When adding a product fields:
- Min quantity
- Delay
- Price min quantity
- Price

Are lost. They haven't been storing any where.

Please fix

Name and code are saving OK.

Thanks

**Comment by sergio-teruel** [comment]
@rafaelbn Fixes done. Can review please?

**Comment by yajo** [comment]
I login as a supplier and go to my account, and I see nothing:

captura de pantalla de 2016-01-07 16-22-22

**Comment by sergio-teruel** [comment]
I check this and your 'proveedor' user is not a supplier.. so you do not see any..

**Comment by yajo** [comment]
:sweat_smile: Thanks!

:+1:

**Comment by carlosdauden** [comment]
:+1:

**Comment by rafaelbn** [comment]
Runbot fails

**Comment by sergio-teruel** [comment]
@rafaelbn runbot fixed...

**Comment by rafaelbn** [comment]
Tested in local enviroment, :+1: 
Runbot and travis errors are not related.
- Runbot something related with OCA servers
- Travis errors are related to website_sale_options
  This module is ready to be merged

**Pull request merged**